Content.Client/_Starlight/Medical/CrewMonitoring/Brigmedic/BrigmedicCrewMonitoringBoundUserInterface.cs

```diff
@@ -0,0 +1,57 @@
+using Content.Shared.Medical.CrewMonitoring;
+using Robust.Client.UserInterface;
+using System.Linq;
+using Content.Shared.Access;
+using Content.Shared.Access.Components;
+using Content.Shared.Access.Systems;
+using Content.Shared.Containers.ItemSlots;
+using Robust.Client.UserInterface;
+using Robust.Shared.Prototypes;
+using static Content.Shared.Access.Components.AccessOverriderComponent;
+namespace Content.Client.Medical.CrewMonitoring.Brigmedic;
+
+public class BrigmedicCrewMonitoringBoundUserInterface : BoundUserInterface
+{
+    [ViewVariables]
+    protected CrewMonitoringWindow? _menu;
+
+    public BrigmedicCrewMonitoringBoundUserInterface(EntityUid owner, Enum uiKey) : base(owner, uiKey)
+    {
+        _accessOverriderSystem = EntMan.System<SharedAccessOverriderSystem>();
+    }
+    protected readonly SharedAccessOverriderSystem _accessOverriderSystem = default!;
+
+    protected override void Open()
+    {
+        EntityUid? gridUid = null;
+        var stationName = string.Empty;
+
+        if (EntMan.TryGetComponent<TransformComponent>(Owner, out var xform))
+        {
+            gridUid = xform.GridUid;
+
+            if (EntMan.TryGetComponent<MetaDataComponent>(gridUid, out var metaData))
+            {
+                stationName = metaData.EntityName;
+            }
+        }
+
+        _menu = this.CreateWindow<CrewMonitoringWindow>();
+        _menu.Set(stationName, gridUid);
+    }
+
+    protected override void UpdateState(BoundUserInterfaceState state)
+    {
+        base.UpdateState(state);
+        switch (state)
```

Comment:
if(state is not CrewMonitoringState st) ?

Reply:
no need for that, original CMO handheld code also doesnt have this check
![image](https://github.com/user-attachments/assets/a15d06fd-139c-4e68-97d4-cd3350e1d53f)

Reply:
Well, it's better not to change the original code to avoid conflicts, but it could have been simplified in a separate version.
The switch doesn't really make sense here; you could have made it a one-liner if instead of all this.

Reply:
true

Reply:
> Well, it's better not to change the original code to avoid conflicts, but it could have been simplified in a separate version. The switch doesn't really make sense here; you could have made it a one-liner if instead of all this.

maybe will change that in the future cuz i am going to keep messing with this code for one of upcoming prs

Reply:
Moreover, the filtering happens only on the front end, so fetching and processing data from the server is pointless. With a modified client, you can display everyone.
But overall, I'm too lazy to make everything be redone.

Reply:
yeah, but there are lots of things that modified client can unlock. Even though it opens door for hackers, wizden decides to keep some logic on client side just to not load server with it

Reply:
In a good implementation, the old state should have been made virtual, inheriting from the brigmed version to avoid duplication, and only transmitting the necessary lists on the server.
Then, on the client side, nothing would need to be changed in the interface, and you could use the same one.

Reply:
> In a good implementation, the old state should have been made virtual, inheriting from the brigmed version to avoid duplication, and only transmitting the necessary lists on the server.
> Then, on the client side, nothing would need to be changed in the interface, and you could use the same one.

thx, will think about that

Reply:
> Some logic on the client side just to avoid overloading the server with it.
But this solution actually creates unnecessary calculations on the server that could have been avoided,
and would slightly reduce the traffic.
Yes, it's a negligible difference that doesn't have a major impact, but technically, it's a more correct solution from this side of the issue.

Reply:
> > But this solution actually creates unnecessary calculations on the server that could have been avoided,
> > and would slightly reduce the traffic.
> > Yes, it's a negligible difference that doesn't have a major impact, but technically, it's a more correct solution from this side of the issue.

alright, alright, i ll fix that someday

Reply:
It's not necessary to fix it, I'm saying it's not important, I'm just pointing it out for the future.